cmd/libsnap-confine-private/string-utils.c

```diff
@@ -99,6 +99,34 @@ size_t sc_string_append(char *dst, size_t dst_size, const char *str)
 	return strlen(dst);
 }
 
+size_t sc_string_append_char(char *dst, size_t dst_size, int c)
+{
+	// Set errno in case we die.
+	errno = 0;
+	if (dst == NULL) {
+		die("cannot append character: buffer is NULL");
+	}
+	size_t dst_len = strnlen(dst, dst_size);
+	if (dst_len == dst_size) {
+		die("cannot append character: dst is unterminated");
+	}
+	size_t max_str_len = dst_size - dst_len;
+	if (max_str_len < 2) {
+		die("cannot append character: not enough space");
+	}
+	if (c < 0 || c > 255) {
```

Comment:
This is puzzling to me for three reasons:

1. `char` is signed in the arches we care about (it's arch-dependent, but I don't know one with unsigned chars). 
2. Why isn't the function itself taking a `char` directly?
3. Are we doing something other than appending ASCII? If not, why not check for the ASCII range, which is 7 bits?